Add gearstats addon

coldmix opened the pull request:
Add gearstats addon to quickly add all the visible stats on gear for quick comparison.

This will print the combined gear stats to the chat window or to a file.

coldmix committed:
Add gearstats addon

Add gearstats addon to quickly add all the visible stats on gear for quick comparison.

RubenatorX commented:
I may have more time to look into this but, my up front questions are:
1) did you author this addon?
2) this sounds up front like it's a lot like the existing addon `checkparam` addon and we'd really rather not have multiple addons doing mostly the same thing when we can help it. We'd typically rather update the existing addon with new features. So what does this do differently than `checkparam`

coldmix commented:
1. Yes, I wrote this addon
2. This just aggregate the numbers on the equipment for a quick comparison. The ideal is to collect all the stats provided by the equipped gear and print them in a simple table for quick comparison of values in game, rather than using a spreadsheet.
For example, I dump the stats twice while switching a piece of gear on a TP gear, this allows a quick comparison of the stats affected (DT, Double Atk, Store TP, Sets etc).
It doesn't consider the default stats from the character job and attributes (i.e. WAR default Double Attack 28% at Master).

```
=====[ TP ]======
 DMG:336 Delay:480 DEF:717 HP:501 MP:53 Haste:26
 STR:203 DEX:131 AGI:115 VIT:201 INT:108 MND:138 CHR:157
 Acc:309 Atk:256 R.Acc:0 R.Atk:10 M.Acc:149 M.Dmg:155
 Eva:298 M.Eva:464 M.Def.Bn:30 DT:-36 PDT:-5
 Store TP:54 Double Atk.:43 Triple Atk.:2 Quad Atk.:3
 TP Bonus:500 PDL:6
 Subtle Blow:18 Subtle Blow II:5
:Skill - Magic Accuracy:228 Great Axe:285 Parrying:269
 Weapon skill DEX:10 Regen:2 Upheaval:1 Crit.hit rate:4 Double Attack damage:20
 Berserk effect duration:15 Blood Rage effect duration:36
Right ear - Double Atk.:8 Subtle Blow:6
Aftermath - Ultimate Skillchain:1 Increases magic burst potency:1
Aftermath - Increases skillchain potency:1
Unity Ranking - Atk:15
Set - Increases Accuracy, Ranged Accuracy, and Magic Accuracy:2
Set - Augments Double Attack:1
```

```
=====[ TP 2 ]======
 DMG:336 Delay:480 DEF:716 HP:481 MP:53 Haste:28
 STR:205 DEX:134 AGI:107 VIT:196 INT:108 MND:132 CHR:148
 Acc:313 Atk:266 R.Acc:0 R.Atk:10 M.Acc:199 M.Dmg:155
 Eva:321 M.Eva:484 M.Def.Bn:31 DT:-45 PDT:-5
 Store TP:50 Double Atk.:34 Triple Atk.:2 Quad Atk.:3
 TP Bonus:500 PDL:6
 Subtle Blow:18 Subtle Blow II:5
:Skill - Magic Accuracy:228 Great Axe:285 Parrying:269
 Weapon skill DEX:10 Regen:2 Upheaval:1 Crit.hit rate:4 Double Attack damage:20
 Critical hit damage:12 Berserk effect duration:15 Blood Rage effect duration:36
 Retaliation:28
Right ear - Double Atk.:8 Subtle Blow:6
Aftermath - Ultimate Skillchain:1 Increases magic burst potency:1
Aftermath - Increases skillchain potency:1
Unity Ranking - Atk:15
Set - Increases Accuracy, Ranged Accuracy, and Magic Accuracy:1
Set - Augments Double Attack:2
```

* Edit, not sure why the line breaks are not showing up correctly in the quotes, see the readme file for stats printout.

coldmix commented:
To give an example of the use case, when I pick up a new piece of gear, I will load the gearswap current set and print the stats, then replace the slot with the new gear and print the stats again.
This will give me a quick assessment to decide whether to replace the gear in gearswap set or not.

There could be multiple sets (high acc, high attack, WS set, etc) to compare so this is a quick approach to go through the effects of the piece on different sets.

RubenatorX commented:
Sounds very similar to what `checkparam` does, except with all stats instead of a list.

coldmix committed:
Add diff and equip set printing

Add command  'base' to set equipment set as base
Add command 'diff' to compare current set with base
Add command 'filediff' to output diff to file
Add printing of equipment set and flag to enable/disable equip printing
Add handling for "physical damage taken:" effects which clash with PDT number.

coldmix committed:
Update README.md

Fix typo for command

coldmix commented:
- Added a change to do diff based on a base set of equipment
- Also print the equip set and the change in gear in the diff.

Example below shows the output
```
//gearstats print
//gearstats base
/equip feet "Boii Calligae +2" 
//gearstats diff
```
Replace `print` with `file` and `diff` with `filediff` to print output to file. diff will only print the changed pieces of gear.
```
=====[ TP Set ]======
main="Chango",sub="Utu Grip",range="Antitail +1",head="Hjarrandi Helm",body="Boii Lorica +2",hands="Sakpata's Gauntlets",legs="Pumm. Cuisses +3",feet="Pumm. Calligae +3",neck="Ainia Collar",left_ear="Telos Earring",right_ear="Boii Earring +1",left_ring="Chirich Ring +1",right_ring="Niqmaddu Ring",waist="Sailfi Belt +1",back="Cichol's Mantle"
 DMG:407 Delay:732 DEF:717 HP:541 MP:93 Haste:26
 STR:198 DEX:126 AGI:115 VIT:196 INT:108 MND:138 CHR:157
 Acc:309 Atk:249 R.Acc:21 R.Atk:31 M.Acc:149 M.Dmg:155
 Eva:298 M.Eva:464 M.Def.Bn:30 DT:-36 PDT:-5
 Store TP:50 Double Atk:43 Triple Atk:2 Quad Atk:3
 TP Bonus:500 PDL:6
 Subtle Blow:18 Subtle Blow II:5
Skill - Magic Accuracy:228 Great Axe:285 Parrying:269 Throwing:242
 Weapon skill DEX:10 Regen:2 Upheaval:1 Double Attack damage:20
 Blood Rage effect duration:36 Crit.hit rate:4 Berserk effect duration:15
Right ear - Double Atk:8 Subtle Blow:6
Aftermath - Ultimate Skillchain:1 Increases magic burst potency:1
Aftermath - Increases skillchain potency:1
Set - Increases Accuracy, Ranged Accuracy, and Magic Accuracy:2
Set - Augments Double Attack:1
Unity Ranking - Atk:15 Double Atk:3
=====[ TP Set 2 ]======
feet="Boii Calligae +2"
 DEF:-1 HP:-20 Haste:2
 STR:2 DEX:3 AGI:-8 VIT:-5 MND:-6 CHR:-9
 Acc:4 Atk:10 M.Acc:50
 Eva:23 M.Eva:20 M.Def.Bn:1 DT:-9
 Store TP:-4 Double Atk:-9
 Critical hit damage:12 Retaliation:28
Set - Increases Accuracy, Ranged Accuracy, and Magic Accuracy:-1
Set - Augments Double Attack:1
```
